thrimshim/thrimshim/main.py

```diff
@@ -236,28 +236,30 @@ def get_row(ident):
 	if is_archive:
 		response["allow_holes"] = True
 
-	# pick default thumbnail template based on start time.
-	# pick default frame time as the middle of the video.
-	# ignore both if video has no start time yet.
-	# or for archive events, no thumbnail.
-	DEFAULT_TEMPLATES = [
-		"zeta-right",
-		"dawnguard-right",
-		"alphaflight-right",
-		"nightwatch-right",
-	]
 	if is_archive:
 		response["thumbnail_mode"] = "NONE"
 	elif response['event_start'] is not None:
 		start = response['event_start']
-		if response['thumbnail_template'] is None:
-			# RDPs default to the RDP template. Others use the current shift.
-			if response['category'] == "RDP":
-				response['thumbnail_template'] = "rdp"
-			else:
-				pst_hour = (start.hour - 8) % 24
-				shift = int(pst_hour / 6)
-				response['thumbnail_template'] = DEFAULT_TEMPLATES[shift]
+
+		# use tags to determine default thumbnail template
+		if response['thumbnail_template'] is None
+			conn = app.db_manager.get_conn()
+			query = """
+				SELECT name, default_template
+				FROM playlists
```

Comment:
Also mentioned below, we want tag instead of name and to exclude nulls here.
We should also exclude any playlist without exactly one tag.
Putting it all together:
```suggestion
				SELECT tags[1] as tag, default_template
				FROM playlists
				WHERE cardinality(tags) = 1 AND default_template IS NOT NULL
```

Reply:
I think my new solution of looping over the tag list of each playlist when creating the `default_templates` dictionary gracefully handles both playlists without any tags and ones with multiple tags.